### PR TITLE
enh(resources): handle severities in the listing

### PR DESCRIPTION
**Pull request by adr-mo**
This PR intends to add

- A new `severity` key to the API response of Resource Status listing
- 4 new filters: service_severity_names, host_severity_names, service_severity_levels, host_severity_levels

API response

image

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Will be fully tested when merged with frontend

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: doc/API/centreon-api-v22.10.yaml [hunk 1]
```diff
@@ -1285,8 +1285,15 @@ paths:
         - $ref: '#/components/parameters/ResourceFilterStatus'
         - $ref: '#/components/parameters/ResourceFilterHostgroupName'
         - $ref: '#/components/parameters/ResourceFilterServicegroupName'
+<<<<<<< HEAD
```
Comment (a-launois) [inline doc/API/centreon-api-v22.10.yaml]: unresolved conflict

File: doc/API/centreon-api-v22.10.yaml [hunk 2]
```diff
@@ -4165,11 +4177,38 @@ components:
       name: host_category_names
       required: false
       description: "Filter the resources by host category names"
+        example: '["High", "Low"]'
```
Comment (a-launois) [inline doc/API/centreon-api-v22.10.yaml]: missing schema ?
and there is two exemples L4190/4191

Comment (adr-mo) [reply doc/API/centreon-api-v22.10.yaml]: Yep rebase with latest merge on categories was quite hard to handle fixing this 👍

File: doc/API/centreon-api-v22.10.yaml (hunk 1 above)
Comment (adr-mo) [reply doc/API/centreon-api-v22.10.yaml]: Yep rebase with latest merge on categories was quite hard to handle fixing this 👍

File: doc/API/centreon-api-v22.10.yaml [hunk 3]
```diff
@@ -4170,6 +4174,46 @@ components:
         items:
           type: string
         example: '["Linux", "Windows"]'
+    ResourceFilterServiceSeverityName:
+      in: query
+      name: service_severity_names
+      required: false
+      description: "Filter the resources by service severity names (only functional with optimized mod)"
```
Comment (adr-mo) [inline doc/API/centreon-api-v22.10.yaml]:
```suggestion
      description: "Filter the resources by service severity names"
```

File: doc/API/centreon-api-v22.10.yaml [hunk 4]
```diff
@@ -4170,6 +4174,46 @@ components:
...
+      schema:
+        type: array
+        items:
+          type: string
+        example: '["Ping", "CPUs"]'
+    ResourceFilterHostSeverityName:
+      in: query
+      name: host_severity_names
+      required: false
+      description: "Filter the resources by host severity names (only functional with optimized mod)"
```
Comment (adr-mo) [inline doc/API/centreon-api-v22.10.yaml]: same

File: doc/API/centreon-api-v22.10.yaml [hunk 5]
```diff
@@ -4170,6 +4174,46 @@ components:
...
+      schema:
+        type: array
+        items:
+          type: string
+        example: '["Linux", "Windows"]'
+    ResourceFilterHostSeverityLevel:
+      in: query
+      name: host_severity_levels
+      required: false
+      description: "Filter the resources by host severity levels (only functional with optimized mod)"
```
Comment (adr-mo) [inline doc/API/centreon-api-v22.10.yaml]: same

File: doc/API/centreon-api-v22.10.yaml [hunk 6]
```diff
@@ -4170,6 +4174,46 @@ components:
...
+      schema:
+        type: array
+        items:
+          type: integer
+        example: '[50, 75]'
+    ResourceFilterServiceSeverityLevel:
+      in: query
+      name: service_severity_levels
+      required: false
+      description: "Filter the resources by service severity levels (only functional with optimized mod)"
```
Comment (adr-mo) [inline doc/API/centreon-api-v22.10.yaml]: same

File: src/Centreon/Infrastructure/Monitoring/Resource/DbReadResourceRepository.php [hunk 7]
```diff
@@ -134,10 +141,8 @@ public function filterByAccessGroups(?array $accessGroups): ResourceRepositoryIn
      */
     public function findResources(ResourceFilter $filter): array
     {
-        $resources = [];
```
Comment (adr-mo) [inline src/Centreon/Infrastructure/Monitoring/Resource/DbReadResourceRepository.php:137]: initialize $this->resources to an empty array

**Review by callapa — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Pull request merged**